--- a/server/Main.cpp
+++ b/server/Main.cpp
@@ -27,6 +27,7 @@
 #include "Debug.h"
 #include <string>
 #include <iostream>
+#include <cstdio>
 #include <signal.h>
 #include <sys/resource.h>
 #include <unistd.h>
--- a/src/Debug.cpp
+++ b/src/Debug.cpp
@@ -24,6 +24,7 @@
 #include <string>
 #include <cstdarg>
 #include <iostream>
+#include <cstdio>
 
 int Debug(unsigned int flags, const char* format, ...)
 {
