From 09acbfde5e5db61f1c0e395c6c52959214635fc5 Mon Sep 17 00:00:00 2001
From: Romain Bignon <romain@budget-insight.com>
Date: Tue, 10 Jun 2014 17:02:15 +0200
Subject: [PATCH] use ConfigParser in priority with python2

---
 weboob/core/repositories.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/weboob/core/repositories.py b/weboob/core/repositories.py
index 23324c9..fa04beb 100644
--- a/weboob/core/repositories.py
+++ b/weboob/core/repositories.py
@@ -40,9 +40,9 @@ from weboob.tools.misc import to_unicode
 from weboob.tools.browser2.browser import BaseBrowser, Weboob as WeboobProfile
 from requests.exceptions import HTTPError
 try:
-    from configparser import RawConfigParser, DEFAULTSECT
-except ImportError:
     from ConfigParser import RawConfigParser, DEFAULTSECT
+except ImportError:
+    from configparser import RawConfigParser, DEFAULTSECT
 
 
 __all__ = ['IProgress', 'ModuleInstallError', 'ModuleInfo', 'RepositoryUnavailable',
-- 
1.8.5.5

