From 667cda095415365cd5e85e4b31914fb4aaa1b7f5 Mon Sep 17 00:00:00 2001
From: Laurent Bachelier <laurent@bachelier.name>
Date: Wed, 21 May 2014 23:23:28 +0200
Subject: [PATCH] Fix Python 2.7.7 breakage

---
 weboob/tools/browser/browser.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/weboob/tools/browser/browser.py b/weboob/tools/browser/browser.py
index 85552a5..af82e58 100644
--- a/weboob/tools/browser/browser.py
+++ b/weboob/tools/browser/browser.py
@@ -735,7 +735,7 @@ class HTTPSConnection2(httplib.HTTPSConnection):
     _HOSTS = {}
     _PROTOCOLS = [ssl.PROTOCOL_TLSv1, ssl.PROTOCOL_SSLv3]
 
-    def _create_connection(self):
+    def _my_create_connection(self):
         sock = socket.create_connection((self.host, self.port), self.timeout)
         if self._tunnel_host:
             self._tunnel()
@@ -746,7 +746,7 @@ class HTTPSConnection2(httplib.HTTPSConnection):
 
     def connect(self):
         for proto in self._get_protocols():
-            sock = self._create_connection()
+            sock = self._my_create_connection()
             try:
                 self.sock = ssl.wrap_socket(sock, self.key_file, self.cert_file, ssl_version=proto)
                 self._HOSTS['%s:%s' % (self.host, self.port)] = [proto]
-- 
1.8.5.5

